tests/ui/test_search.py

```diff
@@ -53,8 +58,87 @@ def test_sorting_by(base_url, selenium, category, sort_attr):
     for item in items.result_list.extensions:
         if term == item.name:
             detail_page = item.click()
             assert detail_page.is_compatible is False
+
+
+@pytest.mark.nondestructive
+def test_search_suggestion_term_is_higher(base_url, selenium):
+    page = Home(selenium, base_url).open()
+    term = 'Ui-Addon'
+    items = page.search.search_for(term, execute=False)
+    for count, item in enumerate(items):
+        if item.name == term:
+            assert count < 1
+
+
+@pytest.mark.nondestructive
+def test_search_suggestion_term_loads_correct_suggestions(base_url, selenium):
+    page = Home(selenium, base_url).open()
+    term = 'Ui-Addon'
+    items = page.search.search_for(term, execute=False)
+    assert term == items[0].name
+    term = '-Install'
+    items = page.search.search_for(term, execute=False)
+    # Sleep to let autocomplete update.
+    time.sleep(2)
+    assert term in items[0].name
+
+
+@pytest.mark.nondestructive
+def test_special_chars_dont_break_suggestions(base_url, selenium):
+    page = Home(selenium, base_url).open()
+    term = 'Ui-Addon-Install'
+    special_chars = '$%^&*'
+    items = page.search.search_for(term, execute=False)
+    assert term == items[0].name
+    items = page.search.search_for(special_chars, execute=False)
+    assert term in items[0].name
+
+
+@pytest.mark.nondestructive
+def test_capitalization_has_same_suggestions(base_url, selenium):
+    page = Home(selenium, base_url).open()
+    term = 'Ui-Addon-Install'
+    items = page.search.search_for(term, execute=False)
+    assert term == items[0].name
+    items = page.search.search_for(term.capitalize(), execute=False)
+    # Sleep to let autocomplete update.
+    time.sleep(2)
+    assert term == items[0].name
+
+
+@pytest.mark.nondestructive
+def test_esc_key_closes_suggestion_list(base_url, selenium):
+    page = Home(selenium, base_url).open()
+    term = 'Ui-Addon-Install'
+    items = page.search.search_for(term, execute=False)
+    action = ActionChains(selenium)
+    # Send ESC key to browser
+    action.send_keys(Keys.ESCAPE).perform()
+    with pytest.raises(NoSuchElementException):
+        selenium.find_element_by_css_selector(
+            'AutoSearchInput-suggestions-list')
+
+
+@pytest.mark.nondestructive
+def test_long_terms_dont_break_suggestions(base_url, selenium):
+    page = Home(selenium, base_url).open()
+    term = 'Ui-Addon' 
+    additional_term = ' 123456789'
+    page.search.search_for(term, execute=False)
+    items = page.search.search_for(additional_term, execute=False)
+    # Sleep to let autocomplete update.
+    time.sleep(2)
+    assert term in items[0].name
+
+
+@pytest.mark.nondestructive
+def test_blank_search_loads_results_page(base_url, selenium):
+    page = Home(selenium, base_url).open() 
+    results = page.search.search_for('', execute=True)
+    assert results.result_list.extensions[0].name == 'Ui-Addon'
+    assert results.result_list.themes[0].name == 'Ui-Test Theme'
```

Comment:
Given that a blank search should issue all add-ons (more or less, depending on platform filters etc) we should be able to expect a specific count of `results` here.

Also, in all other tests the result of `search_for` is called `items`, here it's called `results` (which, I personally like better).

Reply:
Right, so when ```search_for``` has ```execute``` as True, it will load the search results page. I don't quite know of a way to query for how many add-ons there are at one time, so I just figured I'd test to make sure themes and add-ons loaded.